### ISSUE TITLE
Allow contributor to leave a comment to the task provider

### ISSUE THREAD

**Issue by joostschouppe** [maproulette/maproulette2#314]
Sometimes, it is really useful to get some feedback from the mapper. E.g. you leave them a task to fix, but they notice something related to the challenge is pretty weird. It would be nice if there was a free text field where the mapper can leave a message like this. That way, a task manager can more easily have a closer look at fixed-but-special tasks.

**Comment by mvexel** [comment]
I like the idea but it would be a fair amount of work. I wonder if there is a simpler way to accomplish this, for example each task could have a link to send a message to the challenge author?

**Comment by joostschouppe** [comment]
Hmm, that workaround doesn't seem worth the effort: one could simply ask to do so in the instructions. The idea behind this feature request is to create some useful metadata about the individual situations themselves. Given the size of the tasks I have in mind, doing this by individual message would be quite cumbersome.
I wouldn't mind to keep this issue open as a long term goal instead of a quick half-solution.

**Comment by mvexel** [comment]
It would amount to a comment function on tasks (because if you let one mapper comment, the next mapper coming along may want to weigh in as well). I like the idea of it being public but it would also mean some design & implementation work. Let's keep it open for now.

Another 'hacky' solution in the mean time could be to have a link to a github repository you set up for the challenge and pre-fill the title or body with the task ID or a link to the task, like `https://github.com/user/repo/issues/new?title=your+title&body=your+body`. Obviously this would still not show up with the task but perhaps it's something to work with for now?

**Comment by joostschouppe** [comment]
Yes, that sounds like an interesting solution. It would be good enough to help evaluate if it makes sense to fully implement this.

**Comment by joostschouppe** [comment]
This is now functional and looking BEAUTIFUL.
I'm not clear on how it is possible to extract the data though?

**Comment by mvexel** [comment]
For now you'll have to ask me :) -- in the future we should make it
possible to download challenge data as CSV or something.

**Comment by joostschouppe** [comment]
Does that mean that the API documentation is ahead of implementation? (because I see this there:http://maproulette.org/api/v2/task/[task number]/comments

**Comment by mvexel** [comment]
You are right, retrieving via the API is already possible, just not in bulk
for all tasks in a challenge, as far as I remember.
